src/room-terrain.ts

```diff
@@ -9,6 +9,12 @@ interface RoomTerrain {
      * @return number Number of terrain mask like: TERRAIN_MASK_SWAMP | TERRAIN_MASK_WALL
      */
     get(x: number, y: number): 0 | TERRAIN_MASK_WALL | TERRAIN_MASK_SWAMP;
+    /**
+     * Get copy of underlying static terrain buffer.
+     * @param destinationArray (optional) A typed array view in which terrain will be copied to.
+     * @return Uint8Array Copy of underlying room terrain as a new Uint8Array typed array of size 2500.
```

Comment:
Although the documentation says that this method may return `ERR_INVALID_ARGS`, I do not see it in the source. So it's okay to ignore it. Anyway, maybe we should leave a comment about it here.

Reply:
Also, how about adding a jsdoc comment about possible error here
```     
* @throws {RangeError} if `destinationArray` is provided, it must have a length of at least 2500 (`50*50`).
```

Reply:
Thanks for the in-depth review.  I'll update the branch tomorrow and let you know when it's ready for another review.